cblite/llm/LLMProvider.cc

```diff
@@ -30,12 +30,24 @@ using namespace std;
 using namespace fleece;
 using namespace litecore;
 
-LLMProvider* LLMProvider::instance;
-
-fleece::alloc_slice LLMProvider::run(const std::string& restBody, C4Error error, unique_ptr<litecore::REST::Response>& r) {
+fleece::alloc_slice LLMProvider::runSubclass(const std::string& restBody, C4Error error, unique_ptr<litecore::REST::Response>& r) {
     auto headers = getHeaders(restBody);
```

Comment:
This is a bit of awkward back and forth.  I think you can set the headers and body in the subclass, remove the getHeaders method, and just have this method take the response pointer as an argument and nothing else.  Basically, the subclass sets up the response, and the parent class sends it.  Clear separation of responsibility.